dso-l2/src/main/java/com/tc/objectserver/impl/DistributedObjectServer.java

```diff
@@ -820,11 +823,29 @@ public void handleEvent(ReplicationMessageAck context) throws EventHandlerExcept
     final CallbackOnExitHandler handler = new CallbackGroupExceptionHandler(logger, consoleLogger);
     this.threadGroup.addCallbackOnExitExceptionHandler(GroupException.class, handler);
 
+    startDiagnosticListener();
+
+    if (configSetupManager.safeModeStartup()) {
+      try {
+        consoleLogger.warn("Waiting in suspended mode");
+        safeMode.await();
+      } catch (InterruptedException e) {
+        throw new RuntimeException("Startup interrupted while waiting in safe mode");
+      }
+    }
+
     startGroupManagers();
```

Comment:
We may want to put condition here. In some cases, we may want to restart the server